public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -73,7 +73,11 @@ function adjustTargetsFromResponseState(targets: LokiQuery[], response: DataQuer
         maxLines: updatedMaxLines < 0 ? 0 : updatedMaxLines,
       };
     })
-    .filter((target) => target.maxLines === undefined || target.maxLines > 0);
+    .filter(
+      (target) =>
+        (isLogsQuery(target.expr) && (target.maxLines === undefined || target.maxLines > 0)) ||
+        !isLogsQuery(target.expr)
+    );
```

Comment:
Adding it here works and it's actually very clever! The drawback is that this function was never meant to filter queries (based on X or Y criteria), but rather only to make sure we respect the line limit for log queries. Additionally, this function runs on every `nextRequest()` cycle, so it will send the query through the parser in `isLogsQuery()` for every chunk.

If I had to add this to the query splitting module, I would do it within `runSplitQuery()`. There we should do all the filtering, grouping, exclusion, etc, of targets in preparation for `runSplitGroupedQueries()`, which shoul not know about this.

Reply:
Good catch, moved to within the `rangePartitionedMetricQueries` loop, which is already partitioned by isLogsQuery, so no additional tree traversals!